parser/src/string.rs

```diff
@@ -528,6 +528,7 @@ impl<'a> StringParser<'a> {
 
     fn parse_bytes(&mut self) -> Result<Expr, LexicalError> {
         let mut content = String::new();
+        let location = self.get_pos();
```

Comment:
Nit: What would you think of renaming `location` to `start_location` (same in other places)? I would find it useful because it encodes the kind of location.